app/uk/gov/hmrc/eusubsidycompliancefrontend/controllers/UndertakingController.scala

```diff
@@ -281,6 +286,49 @@ class UndertakingController @Inject() (
     }
   }
 
+  def getDisableUndertakingWarning: Action[AnyContent] = withCDSAuthenticatedUser.async { implicit request =>
+    withLeadUndertaking(undertaking => Ok(disableUndertakingWarningPage(undertaking.name.toString)).toFuture)
+  }
+
+  def getDisableUndertakingConfirm: Action[AnyContent] = withCDSAuthenticatedUser.async { implicit request =>
+    withLeadUndertaking(undertaking =>
+      Ok(disableUndertakingConfirmPage(disableUndertakingConfirmForm, undertaking.name)).toFuture
+    )
+  }
+
+  def postDisableUndertakingConfirm: Action[AnyContent] = withCDSAuthenticatedUser.async { implicit request =>
+    withLeadUndertaking { undertaking =>
+      disableUndertakingConfirmForm
+        .bindFromRequest()
+        .fold(
+          errors => BadRequest(disableUndertakingConfirmPage(errors, undertaking.name)).toFuture,
+          form => getNext(form, undertaking)
+        )
+    }
+  }
+
+  def getUndertakingDisabled: Action[AnyContent] = withCDSAuthenticatedUser.async { implicit request =>
+    Ok(undertakingDisabledPage()).toFuture
+  }
+
+  private def resetAllJourneys(implicit eori: EORI) =
+    for {
+      _ <- store.delete[EligibilityJourney]
+      _ <- store.delete[UndertakingJourney]
+      _ <- store.delete[NewLeadJourney]
+      _ <- store.delete[NilReturnJourney]
+      _ <- store.delete[BusinessEntityJourney]
+      _ <- store.delete[BecomeLeadJourney]
+      _ <- store.delete[SubsidyJourney]
+    } yield ()
+
+  def getNext(form: FormValues, undertaking: Undertaking)(implicit hc: HeaderCarrier) = if (form.value == "true") {
```

Comment:
Minor - Consider renaming this to something that reflects what it does better like `handleFormSubmission`.